diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6613972..a847aac 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,6 +61,9 @@ dolce_create_self(${PROJECT_NAME}.suprx ${PROJECT_NAME}
 dolce_create_stubs(${PROJECT_NAME}_stubs 
   ${PROJECT_NAME} 
   ${CMAKE_SOURCE_DIR}/exports.yml
-  LIB vitaSAS_stub
+  LIB vitaSAS_stub vitaSAS_stub_weak
 )
 
+install(FILES ${CMAKE_SOURCE_DIR}/include/vitaSAS.h DESTINATION include)
+install(FILES $<TARGET_FILE:vitaSAS_stub> $<TARGET_FILE:vitaSAS_stub_weak> DESTINATION lib)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.suprx DESTINATION module)
diff --git a/Makefile b/Makefile
index d75d998..8c64136 100644
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@ INCLUDES   = include
 PREFIX  ?= ${DOLCESDK}/arm-dolce-eabi
 CC      = arm-dolce-eabi-gcc
 AR      = arm-dolce-eabi-ar
-CFLAGS  = -Wl,-q -Wall -O3 -I$(INCLUDES) -ffat-lto-objects -flto
+CFLAGS  = -Wl,-q -Wall -O3 -I$(INCLUDES)
 ASFLAGS = $(CFLAGS)
 
 all: $(TARGET_LIB)
